### DISCUSSION TITLE
inject clink into fork's git instance

### DISCUSSION THREAD

**Discussion by VPraharsha03** [chrisant996/clink#670] [Q&A] [answered]
I use fork client on windows that uses it's very own built-in instance (git for windows underhood), how can I inject clink everytime I launch the CLI from gui client? 
to enjoy the completion features

**Comment by chrisant996** [comment]
> I use fork client on windows that uses it's very own built-in instance (git for windows underhood), how can I inject clink everytime I launch the CLI from gui client? 
> to enjoy the completion features 

I'm having trouble understanding the question; I think some context needs to be provided.

- What is "fork client"?
- What is meant by "its very own built-in instance (git for Windows underhood)"?
- What does "gui client" refer to?
- What does "the CLI" refer to?
- How does the gui client "launch the CLI"?  Is there a way to configure what the gui client launches?

Does "launch the CLI from gui client" launch an instance of `cmd.exe`, or does it launch something else?

Clink can only be injected into `cmd.exe` instances.

**Comment by VPraharsha03** [comment]
@chrisant996 Sorry for lacking details - I was talking about the git GUI client called Fork (https://git-fork.com/) which comes with it's own git instance (Git for Windows), the CLI called from the GUI client is "bash.exe" (MSYS64) 

I'm not sure if I can configure the client to launch a different process on clicking the CLI icon, the only thing i'm aware is that `cmd.exe` uses autorun to call clink

**Reply by chrisant996** [reply, marked as answer]
Clink injects into cmd.exe.

Bash is bash.exe.  Bash cannot have Clink injected into it.

**Comment by VPraharsha03** [comment]
The only way I could get auto-completion feature and shell history lookup into `bash.exe` was to call a different shell `nu` from `.bashrc` of GFW

**Reply by chrisant996** [reply]
You're looking for help on how to configure bash, and maybe specifically git bash (part of git for windows).

This is the Clink forum, and isn't about bash.

You'll have better luck posting in a forum about bash or git for windows.

**Reply by VPraharsha03** [reply]
No, my actual question was to know whether clink could be injected into processes other than `cmd.exe`, which isn't possible so I went with an easier solution.

**Reply by chrisant996** [reply]
Ok. What I'm trying to convey is that bash is capable of completions, and there are tons more completion scripts for bash than there are for Clink.

You just have to configure bash (just like you have to configure Clink). I suppose nu must come with enough completions pre-configured that you don't need to, or you already configured it.

But if you're willing to launch nu from bash, you could launch cmd from bash, if what you want is specifically Clink completions.

**Reply by VPraharsha03** [reply]
Yes, usual way would be to look for completion scripts for bash, but this seemed to be a more easier solution as i'd nushell pre-installed.

Thanks a lot for the suggestions!
